<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Similarity and Distance Quantification Between Probability Functions â€¢ philentropy</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Similarity and Distance Quantification Between Probability Functions">
<meta property="og:description" content="Computes 46 optimized distance and similarity measures for comparing probability functions (Drost (2018) &lt;doi:10.21105/joss.00765&gt;). These comparisons between probability functions have their foundations in a broad range of scientific disciplines from mathematics to ecology. The aim of this package is to provide a core framework for clustering, classification, statistical inference, goodness-of-fit, non-parametric statistics, information theory, and machine learning tasks that are based on comparing univariate or multivariate probability functions.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">philentropy</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.10.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/Distances.html">Distances</a>
    </li>
    <li>
      <a href="articles/Information_Theory.html">Information Theory</a>
    </li>
    <li>
      <a href="articles/Introduction.html">Introduction</a>
    </li>
    <li>
      <a href="articles/Many_Distances.html">Comparing many probability density functions</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drostlab/philentropy/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="philentropy--information-theory-and-distance-quantification-with-r">philentropy <sub><sup>â€” Information Theory and Distance Quantification with R</sup></sub><a class="anchor" aria-label="anchor" href="#philentropy--information-theory-and-distance-quantification-with-r"></a>
</h1></div>
<p><a href="https://cran.r-project.org/package=philentropy" class="external-link"><img src="https://img.shields.io/cran/v/philentropy" alt="CRAN"></a> <a href="https://joss.theoj.org/papers/10.21105/joss.00765" class="external-link"><img src="http://joss.theoj.org/papers/cad5ffc246ce197b06ccad1af7d2932a/status.svg" alt="status"></a> <a href="https://github.com/r-hub/cranlogs.app" class="external-link"><img src="https://cranlogs.r-pkg.org/badges/philentropy" alt="rstudio mirror downloads"></a> <a href="https://github.com/r-hub/cranlogs.app" class="external-link"><img src="https://cranlogs.r-pkg.org/badges/grand-total/philentropy" alt="rstudio mirror downloads"></a></p>
<div class="section level2">
<h2 id="compass-similarity-and-distance-quantification-between-probability-functions">ğŸ§­ Similarity and Distance Quantification between Probability Functions<a class="anchor" aria-label="anchor" href="#compass-similarity-and-distance-quantification-between-probability-functions"></a>
</h2>
<blockquote>
<p><em>Describe and understand the world through data.</em></p>
</blockquote>
<p>Data collection and data comparison are the foundations of scientific research.<br><em>Mathematics</em> provides the abstract framework to describe patterns we observe in nature and <em>Statistics</em> provides the framework to quantify the uncertainty of these patterns.</p>
<p>In statistics, natural patterns are described in the form of probability distributions that either follow fixed patterns (parametric distributions) or more dynamic ones (non-parametric distributions).</p>
<p>The <code>philentropy</code> package implements fundamental distance and similarity measures to quantify distances between probability density functions as well as traditional information theory measures.<br>
In this regard, it aims to provide a framework for comparing natural patterns in a statistical notation.</p>
<blockquote>
<p>ğŸ§¡ This project is born out of my passion for statistics and I hope it will be useful to those who share it with me.</p>
</blockquote>
<hr>
</div>
<div class="section level2">
<h2 id="gear-installation">âš™ï¸ Installation<a class="anchor" aria-label="anchor" href="#gear-installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install philentropy version 0.10.0 from CRAN</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"philentropy"</span><span class="op">)</span></span></code></pre></div>
<p>Or get the latest developer version:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"HajkD/philentropy"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<hr>
</div>
<div class="section level2">
<h2 id="receipt-citation">ğŸ§¾ Citation<a class="anchor" aria-label="anchor" href="#receipt-citation"></a>
</h2>
<blockquote>
<p>HG Drost (2018).<br><strong>Philentropy: Information Theory and Distance Quantification with R</strong>.<br><em>Journal of Open Source Software</em>, 3(26), 765.<br><a href="https://doi.org/10.21105/joss.00765" class="external-link uri">https://doi.org/10.21105/joss.00765</a></p>
</blockquote>
<blockquote>
<p>ğŸª¶ <em>I am developing <code>philentropy</code> in my spare time and would be very grateful if you would consider citing the paper above if it was useful for your research. These citations help me continue maintaining and extending the package.</em></p>
</blockquote>
<hr>
</div>
<div class="section level2">
<h2 id="jigsaw-quick-start">ğŸ§© Quick Start<a class="anchor" aria-label="anchor" href="#jigsaw-quick-start"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drostlab.github.io/philentropy/" class="external-link">philentropy</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">P</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.2</span>, <span class="fl">0.7</span><span class="op">)</span></span>
<span><span class="va">Q</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.6</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="reference/distance.html">distance</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">P</span>, <span class="va">Q</span><span class="op">)</span>, method <span class="op">=</span> <span class="st">"jensen-shannon"</span><span class="op">)</span></span></code></pre></div>
<pre><code>jensen-shannon using unit 'log'.
jensen-shannon
    0.02628933</code></pre>
<blockquote>
<p>ğŸ’¡ <strong>Tip:</strong> Got a large matrix (rows = samples, cols = features)?<br>
Use <code>distance(X, method="cosine", mute.message=TRUE)</code> to compute the full pairwise matrix quickly and quietly.</p>
</blockquote>
<hr>
</div>
<div class="section level2">
<h2 id="blue_book-tutorials">ğŸ“˜ Tutorials<a class="anchor" aria-label="anchor" href="#blue_book-tutorials"></a>
</h2>
<ul>
<li>
<a href="https://drostlab.github.io/philentropy/articles/Introduction.html" class="external-link">Introduction to the philentropy package</a><br>
</li>
<li>
<a href="https://drostlab.github.io/philentropy/articles/Distances.html" class="external-link">Distance and Similarity Measures implemented in philentropy</a><br>
</li>
<li>
<a href="https://drostlab.github.io/philentropy/articles/Information_Theory.html" class="external-link">Information Theory Metrics implemented in philentropy</a><br>
</li>
<li><a href="https://drostlab.github.io/philentropy/articles/Many_Distances.html" class="external-link">Comparing many probability density functions</a></li>
</ul>
<hr>
</div>
<div class="section level2">
<h2 id="test_tube-when-should-i-use-which-distance">ğŸ§ª When should I use which distance?<a class="anchor" aria-label="anchor" href="#test_tube-when-should-i-use-which-distance"></a>
</h2>
<table class="table">
<colgroup>
<col width="22%">
<col width="77%">
</colgroup>
<thead><tr>
<th>Goal</th>
<th>Recommended Methods</th>
</tr></thead>
<tbody>
<tr>
<td>ğŸ” Clustering / similarity</td>
<td>
<code>cosine</code>, <code>correlation</code>, <code>euclidean</code>
</td>
</tr>
<tr>
<td>ğŸ“Š Probability or compositional data</td>
<td>
<code>jensen-shannon</code>, <code>hellinger</code>, <code>kullback-leibler</code>
</td>
</tr>
<tr>
<td>ğŸ§¬ Sparse counts / binary</td>
<td>
<code>canberra</code>, <code>jaccard</code>, <code>sorensen</code>
</td>
</tr>
<tr>
<td>âš–ï¸ Scale-invariant</td>
<td>
<code>manhattan</code>, <code>chebyshev</code>
</td>
</tr>
</tbody>
</table>
<blockquote>
<p>Run <code><a href="reference/getDistMethods.html">getDistMethods()</a></code> to explore all 50+ implemented measures.</p>
</blockquote>
<hr>
</div>
<div class="section level2">
<h2 id="abacus-examples">ğŸ§® Examples<a class="anchor" aria-label="anchor" href="#abacus-examples"></a>
</h2>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://drostlab.github.io/philentropy/" class="external-link">philentropy</a></span><span class="op">)</span></span>
<span><span class="fu">philentropy</span><span class="fu">::</span><span class="fu"><a href="reference/getDistMethods.html">getDistMethods</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># define probability density functions P and Q</span></span>
<span><span class="va">P</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">Q</span> <span class="op">&lt;-</span> <span class="fl">20</span><span class="op">:</span><span class="fl">29</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fl">20</span><span class="op">:</span><span class="fl">29</span><span class="op">)</span></span>
<span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">P</span>, <span class="va">Q</span><span class="op">)</span></span>
<span><span class="fu">philentropy</span><span class="fu">::</span><span class="fu"><a href="reference/distance.html">distance</a></span><span class="op">(</span><span class="va">x</span>, method <span class="op">=</span> <span class="st">"jensen-shannon"</span><span class="op">)</span></span></code></pre></div>
<pre><code>jensen-shannon using unit 'log'.
jensen-shannon
    0.02628933</code></pre>
<p>Alternatively, compute all available distances:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">philentropy</span><span class="fu">::</span><span class="fu"><a href="reference/dist.diversity.html">dist.diversity</a></span><span class="op">(</span><span class="va">x</span>, p <span class="op">=</span> <span class="fl">2</span>, unit <span class="op">=</span> <span class="st">"log2"</span><span class="op">)</span></span></code></pre></div>
<hr>
</div>
<div class="section level2">
<h2 id="star2-papers-using-philentropy-highlights">ğŸŒŸ Papers using philentropy (highlights)<a class="anchor" aria-label="anchor" href="#star2-papers-using-philentropy-highlights"></a>
</h2>
<blockquote>
<p><em>Flagship examples with top venues. Click to expand full lists.</em></p>
</blockquote>
<details><summary><b>Nature / Cell / Science</b>
</summary><ul>
<li>
<strong>A transcriptomic hourglass in brown algae</strong><br>
JS Lotharukpong, M Zheng, R Luthringer et al.Â â€“ <strong>Nature</strong>, 2024<br>
</li>
<li>
<strong>Annelid functional genomics reveal the origins of bilaterian life cycles</strong><br>
FM MartÃ­n-Zamora, Y Liang, K Guynes et al.Â â€“ <strong>Nature</strong>, 2023<br>
</li>
<li>
<strong>An atlas of gene regulatory elements in adult mouse cerebrum</strong><br>
YE Li, S Preissl, X Hou, Z Zhang, K Zhang et al.Â â€“ <strong>Nature</strong>, 2021<br>
</li>
<li>
<strong>Convergent somatic mutations in metabolism genes in chronic liver disease</strong><br>
S Ng, F Rouhani, S Brunner, N Brzozowska et al.Â â€“ <strong>Nature</strong>, 2021<br>
</li>
<li>
<strong>Antigen dominance hierarchies shape TCF1+ progenitor CD8 T cell phenotypes in tumors</strong><br>
ML Burger, AM Cruz, GE Crossland et al.Â â€“ <strong>Cell</strong>, 2021<br>
</li>
<li>
<strong>A comparative atlas of single-cell chromatin accessibility in the human brain</strong><br>
YE Li, S Preissl, M Miller, ND Johnson, Z Wang et al.Â â€“ <strong>Science</strong>, 2023</li>
</ul></details><details><summary><b>Nature Methods / Nat Comms / Cell family</b>
</summary><ul>
<li>
<strong>sciCSR infers B cell state transition and predicts class-switch recombination dynamics using scRNA-seq</strong><br>
JCF Ng, G Montamat Garcia, AT Stewart et al.Â â€“ <strong>Nature Methods</strong>, 2024<br>
</li>
<li>
<strong>Decoding the gene regulatory network of endosperm differentiation in maize</strong><br>
Y Yuan, Q Huo, Z Zhang, Q Wang, J Wang et al.Â â€“ <strong>Nature Communications</strong>, 2024<br>
</li>
<li>
<strong>Population structure in a fungal human pathogen is potentially linked to pathogenicity</strong><br>
EA Hatmaker, AE Barber, MT Drott et al.Â â€“ <strong>Nature Communications</strong>, 2025<br>
</li>
<li>
<strong>Pan-cancer human brain metastases atlas at single-cell resolution</strong><br>
X Xing, J Zhong, J Biermann, H Duan, X Zhang et al.Â â€“ <strong>Cancer Cell</strong>, 2025<br>
</li>
<li>
<strong>Gene module reconstruction identifies cellular differentiation processes and the regulatory logic of specialized secretion in zebrafish</strong><br>
Y Wang, J Liu, LY Du, JL Wyss, JA Farrell, AF Schier â€“ <strong>Developmental Cell</strong>, 2025<br>
</li>
</ul></details><details><summary><b>Other disciplines (selected)</b>
</summary><ul>
<li>
<strong>Staphylococci in high resolution: Capturing diversity within the human nasal microbiota</strong><br>
AC Ingham, DYK Ng, S Iversen, CM Liu et al.Â â€“ <strong>Cell Reports</strong>, 2025<br>
</li>
<li>
<strong>The power of visualizing distributional differences: formal graphical n-sample tests</strong><br>
K Konstantinou, T MrkviÄka, M MyllymÃ¤ki â€“ <strong>Computational Statistics</strong>, 2025<br>
</li>
<li>
<strong>Plant species as ecological engineers of microtopography in a temperate sedge-grass marsh</strong><br>
J DuÅ¡ek, J NovotnÃ½, B NavrÃ¡tilovÃ¡ et al.Â â€“ <strong>Scientific Reports</strong>, 2025<br>
</li>
<li>
<strong>Resolution of MALDI-TOF vs WGS for Bacillus identification (NASA JSC)</strong><br>
F Mazhari, AB Regberg, CL Castro, MG LaMontagne â€“ <strong>Frontiers in Microbiology</strong>, 2025<br>
</li>
<li>
<strong>Every Hue Has Its Fan Club: Diverse Patterns of Color-Dependent Flower Visitation across Lepidoptera</strong><br>
D Kutcherov, EL Westerman â€“ <strong>Integrative and Comparative Biology</strong>, 2025<br>
</li>
</ul></details><blockquote>
<p>ğŸ“ philentropy has been used in <strong>dozens of peer-reviewed publications</strong> to quantify distances, divergences, and similarities in complex biological and computational datasets.</p>
</blockquote>
<hr>
</div>
<div class="section level2">
<h2 id="brain-important-functions">ğŸ§  Important Functions<a class="anchor" aria-label="anchor" href="#brain-important-functions"></a>
</h2>
<div class="section level3">
<h3 id="distance-measures">Distance Measures<a class="anchor" aria-label="anchor" href="#distance-measures"></a>
</h3>
<ul>
<li>
<code><a href="reference/distance.html">distance()</a></code> â€“ Implements 46 probability distance/similarity measures<br>
</li>
<li>
<code><a href="reference/getDistMethods.html">getDistMethods()</a></code> â€“ Get all method names for <code>distance</code><br>
</li>
<li>
<code><a href="reference/dist.diversity.html">dist.diversity()</a></code> â€“ Computes distance diversity between PDFs<br>
</li>
<li>
<code><a href="reference/estimate.probability.html">estimate.probability()</a></code> â€“ Estimate probability vectors from counts</li>
</ul>
</div>
<div class="section level3">
<h3 id="information-theory">Information Theory<a class="anchor" aria-label="anchor" href="#information-theory"></a>
</h3>
<ul>
<li>
<code><a href="reference/H.html">H()</a></code> â€“ Shannonâ€™s Entropy <code>H(X)</code><br>
</li>
<li>
<code><a href="reference/JE.html">JE()</a></code> â€“ Joint Entropy <code>H(X,Y)</code><br>
</li>
<li>
<code><a href="reference/CE.html">CE()</a></code> â€“ Conditional Entropy <code>H(X|Y)</code><br>
</li>
<li>
<code><a href="reference/MI.html">MI()</a></code> â€“ Mutual Information <code>I(X,Y)</code><br>
</li>
<li>
<code><a href="reference/KL.html">KL()</a></code> â€“ Kullbackâ€“Leibler Divergence<br>
</li>
<li>
<code><a href="reference/JSD.html">JSD()</a></code> â€“ Jensenâ€“Shannon Divergence<br>
</li>
<li>
<code><a href="reference/gJSD.html">gJSD()</a></code> â€“ Generalized Jensenâ€“Shannon Divergence</li>
</ul>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="newspaper_roll-news">ğŸ—ï¸ NEWS<a class="anchor" aria-label="anchor" href="#newspaper_roll-news"></a>
</h2>
<p>Find the current status and version history in the<br>
ğŸ‘‰ <a href="https://drostlab.github.io/philentropy/news/index.html" class="external-link">NEWS section</a>.</p>
<hr>
</div>
<div class="section level2">
<h2 id="jigsaw-appendix--full-references">ğŸ§© Appendix â€” full references<a class="anchor" aria-label="anchor" href="#jigsaw-appendix--full-references"></a>
</h2>
<blockquote>
<ul>
<li><p><strong>A transcriptomic hourglass in brown algae</strong><br>
JS Lotharukpong, M Zheng, R Luthringer et al.Â â€“ <strong>Nature</strong>, 2024</p></li>
<li><p><strong>Annelid functional genomics reveal the origins of bilaterian life cycles</strong><br>
FM MartÃ­n-Zamora, Y Liang, K Guynes et al.Â â€“ <strong>Nature</strong>, 2023</p></li>
<li><p><strong>An atlas of gene regulatory elements in adult mouse cerebrum</strong><br>
YE Li, S Preissl, X Hou, Z Zhang, K Zhang et al.Â â€“ <strong>Nature</strong>, 2021</p></li>
<li><p><strong>Convergent somatic mutations in metabolism genes in chronic liver disease</strong><br>
S Ng, F Rouhani, S Brunner, N Brzozowska et al.Â â€“ <strong>Nature</strong>, 2021</p></li>
<li><p><strong>Antigen dominance hierarchies shape TCF1+ progenitor CD8 T cell phenotypes in tumors</strong><br>
ML Burger, AM Cruz, GE Crossland et al.Â â€“ <strong>Cell</strong>, 2021</p></li>
<li><p><strong>High-content single-cell combinatorial indexing</strong><br>
R Mulqueen et al.Â â€“ <strong>Nature Biotechnology</strong>, 2021</p></li>
<li><p><strong>A comparative atlas of single-cell chromatin accessibility in the human brain</strong><br>
YE Li, S Preissl, M Miller, ND Johnson, Z Wang et al.Â â€“ <strong>Science</strong>, 2023</p></li>
<li><p><strong>Extinction at the end-Cretaceous and the origin of modern Neotropical rainforests</strong><br>
MR Carvalho, C Jaramillo et al.Â â€“ <strong>Science</strong>, 2021</p></li>
<li><p><strong>sciCSR infers B cell state transition and predicts class-switch recombination dynamics using single-cell transcriptomic data</strong><br>
JCF Ng, G Montamat Garcia, AT Stewart et al.Â â€“ <strong>Nature Methods</strong>, 2024</p></li>
<li><p><strong>HERMES: a molecular-formula-oriented method to target the metabolome</strong><br>
R GinÃ©, J Capellades, JM Badia et al.Â â€“ <strong>Nature Methods</strong>, 2021</p></li>
<li><p><strong>Epithelial zonation along the mouse and human small intestine defines five discrete metabolic domains</strong><br>
RK Zwick, P Kasparek, B Palikuqi et al.Â â€“ <strong>Nature Cell Biology</strong>, 2024</p></li>
<li><p><strong>The genetic architecture of temperature adaptation is shaped by population ancestry and not by selection regime</strong><br>
KA Otte, V Nolte, F Mallard et al.Â â€“ <strong>Genome Biology</strong>, 2021</p></li>
<li><p><strong>The Tug1 lncRNA locus is essential for male fertility</strong><br>
JP Lewandowski et al.Â â€“ <strong>Genome Biology</strong>, 2020</p></li>
<li><p><strong>Decoding the gene regulatory network of endosperm differentiation in maize</strong><br>
Y Yuan, Q Huo, Z Zhang, Q Wang, J Wang et al.Â â€“ <strong>Nature Communications</strong>, 2024</p></li>
<li><p><strong>A full-body transcription factor expression atlas with completely resolved cell identities in C. elegans</strong><br>
Y Li, S Chen, W Liu, D Zhao, Y Gao, S Hu, H Liu, Y Li et al.Â â€“ <strong>Nature Communications</strong>, 2024</p></li>
<li><p><strong>Comprehensive mapping and modelling of the rice regulome landscape unveils the regulatory architecture underlying complex traits</strong><br>
T Zhu, C Xia, R Yu, X Zhou, X Xu, L Wang et al.Â â€“ <strong>Nature Communications</strong>, 2024</p></li>
<li><p><strong>Transcriptional vulnerabilities of striatal neurons in human and rodent models of Huntingtonâ€™s disease</strong><br>
A Matsushima, SS Pineda, JR Crittenden et al.Â â€“ <strong>Nature Communications</strong>, 2023</p></li>
<li><p><strong>Population structure in a fungal human pathogen is potentially linked to pathogenicity</strong><br>
EA Hatmaker, AE Barber, MT Drott et al.Â â€“ <strong>Nature Communications</strong>, 2025</p></li>
<li><p><strong>Resolving the structure of phageâ€“bacteria interactions in the context of natural diversity</strong><br>
KM Kauffman, WK Chang, JM Brown et al.Â â€“ <strong>Nature Communications</strong>, 2022</p></li>
<li><p><strong>Gut microbiome-mediated metabolism effects on immunity in rural and urban African populations</strong><br>
M StraÅ¾ar, GS Temba, H Vlamakis et al.Â â€“ <strong>Nature Communications</strong>, 2021</p></li>
<li><p><strong>Aging, inflammation and DNA damage in the somatic testicular niche with idiopathic germ cell aplasia</strong><br>
M Alfano, AS Tascini, F Pederzoli et al.Â â€“ <strong>Nature Communications</strong>, 2021</p></li>
<li><p><strong>Single cell census of human kidney organoids shows reproducibility and diminished off-target cells after transplantation</strong><br>
A Subramanian et al.Â â€“ <strong>Nature Communications</strong>, 2019</p></li>
<li><p><strong>Pan-cancer human brain metastases atlas at single-cell resolution</strong><br>
X Xing, J Zhong, J Biermann, H Duan, X Zhang et al.Â â€“ <strong>Cancer Cell</strong>, 2025</p></li>
<li><p><strong>The temporal progression of lung immune remodeling during breast cancer metastasis</strong><br>
CS McGinnis, Z Miao, D Superville, W Yao et al.Â â€“ <strong>Cancer Cell</strong>, 2024</p></li>
<li><p><strong>Cross-tissue human fibroblast atlas reveals myofibroblast subtypes with distinct roles in immune modulation</strong><br>
Y Gao, J Li, W Cheng, T Diao, H Liu, Y Bo, C Liu et al.Â â€“ <strong>Cancer Cell</strong>, 2024</p></li>
<li><p><strong>Gene module reconstruction identifies cellular differentiation processes and the regulatory logic of specialized secretion in zebrafish</strong><br>
Y Wang, J Liu, LY Du, JL Wyss, JA Farrell, AF Schier â€“ <strong>Developmental Cell</strong>, 2025</p></li>
<li><p><strong>Large-scale chromatin reorganization reactivates placenta-specific genes that drive cellular aging</strong><br>
Z Liu, Q Ji, J Ren, P Yan, Z Wu, S Wang, L Sun, Z Wang et al.Â â€“ <strong>Developmental Cell</strong>, 2022</p></li>
<li><p><strong>Integrated single-cell and spatial transcriptomic profiling reveals that CD177+ Tregs enhance immunosuppression through apoptosis and resistance to â€¦</strong><br>
Y Liang, L Qiao, Q Qian, R Zhang, Y Li, X Xu et al.Â â€“ <strong>Oncogene</strong>, 2025</p></li>
<li><p><strong>Conserved and unique features of terminal telomeric sequences in ALT-positive cancer cells</strong><br>
B Azeroglu, W Wu, R Pavani, RS Sandhu et al.Â â€“ <strong>eLife</strong>, 2025</p></li>
<li><p><strong>Spotless, a reproducible pipeline for benchmarking cell type deconvolution in spatial transcriptomics</strong><br>
C Sang-Aram, R Browaeys, R Seurinck, Y Saeys â€“ <strong>eLife</strong>, 2024</p></li>
<li><p><strong>Loss of adaptive capacity in asthmatic patients revealed by biomarker fluctuation dynamics after rhinovirus challenge</strong><br>
A Sinha et al.Â â€“ <strong>eLife</strong>, 2019</p></li>
<li><p><strong>Staphylococci in high resolution: Capturing diversity within the human nasal microbiota</strong><br>
AC Ingham, DYK Ng, S Iversen, CM Liu et al.Â â€“ <strong>Cell Reports</strong>, 2025</p></li>
<li><p><strong>Triple network dynamics and future alcohol consumption in adolescents</strong><br>
CC McIntyre, M Khodaei, RG Lyday et al.Â â€“ <strong>Alcohol: Clinical and Experimental Research</strong>, 2025</p></li>
<li><p><strong>Benchmarking 13 tools for mutational signature attribution, including a new and improved algorithm</strong><br>
N Jiang, Y Wu, SG Rozen â€“ <strong>Briefings in Bioinformatics</strong>, 2025</p></li>
<li><p><strong>Plant species as ecological engineers of microtopography in a temperate sedge-grass marsh</strong><br>
J DuÅ¡ek, J NovotnÃ½, B NavrÃ¡tilovÃ¡ et al.Â â€“ <strong>Scientific Reports</strong>, 2025</p></li>
<li><p><strong>Resolution of MALDI-TOF compared to whole genome sequencing for identification of Bacillus species isolated from cleanrooms at NASA Johnson Space Center</strong><br>
F Mazhari, AB Regberg, CL Castro, MG LaMontagne â€“ <strong>Frontiers in Microbiology</strong>, 2025</p></li>
<li><p><strong>Every Hue Has Its Fan Club: Diverse Patterns of Color-Dependent Flower Visitation across Lepidoptera</strong><br>
D Kutcherov, EL Westerman â€“ <strong>Integrative and Comparative Biology</strong>, 2025</p></li>
<li><p><strong>An in vivo CRISPR screen in chick embryos reveals a role for MLLT3 in specification of neural cells from the caudal epiblast</strong><br>
ARG Libby, T Rito, A Radley, J Briscoe â€“ <strong>Development</strong>, 2025</p></li>
<li><p><strong>Single-Cell Analyses Reveal a Functionally Heterogeneous Exhausted CD8+ T-cell Subpopulation That Is Correlated with Response to Checkpoint Therapy in â€¦</strong><br>
KM Mahuron, O Shahid, P Sao, C Wu et al.Â â€“ <strong>Cancer Research</strong>, 2025</p></li>
<li><p><strong>ETS1â€Driven Nucleolar Stress Orchestrates OLR1+ Macrophage Crosstalk to Sustain Immunosuppressive Microenvironment in Clear Cell Renal Cell Carcinoma</strong><br>
L Xiao, Z Zhang, T Li, Y Jiang, Y Liu, J Wang, W Tang â€“ <strong>Human Mutation</strong>, 2025</p></li>
<li><p><strong>Association Between Ocular Microbiomes of Children and Their Siblings and Parents</strong><br>
X Ling, Y Zhang, CHT Bui, HN Chan, POS Tam et al.Â â€“ <strong>Investigative Ophthalmology &amp; Visual Science</strong>, 2025</p></li>
<li><p><strong>Benefits and challenges of host depletion methods in profiling the upper and lower respiratory microbiome</strong><br>
C Wang, L Zhang, C Kan, J He, W Liang, R Xia et al.Â â€“ <strong>Biofilms and Microbiomes</strong>, 2025</p></li>
<li><p><strong>Unsettled Times: Music Discovery Reveals Divergent Cultural Responses to War</strong><br>
H Lee, M Anglada-Tort, O Sobchuk et al.Â â€“ <strong>PsyArXiv Preprints</strong>, 2025</p></li>
<li><p><strong>q-Generalization of Nakagami distribution with applications</strong><br>
N Kumar, A Dixit, V Vijay â€“ <strong>Japanese Journal of Statistics and Data Science</strong>, 2025</p></li>
<li><p><strong>The power of visualizing distributional differences: formal graphical n-sample tests</strong><br>
K Konstantinou, T MrkviÄka, M MyllymÃ¤ki â€“ <strong>Computational Statistics</strong>, 2025</p></li>
<li><p><strong>Topic Modeling of Positive and Negative Reviews of Soulslike Video Games</strong><br>
T Guzsvinecz â€“ <strong>Computers</strong>, 2025</p></li>
<li><p><strong>Basic Statistical Inference</strong><br>
M Nguyen â€“ <strong>Foundations of Data Analysis</strong>, 2025</p></li>
</ul>
</blockquote>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=philentropy" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/drostlab/philentropy/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/drostlab/philentropy/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-2" class="external-link">GPL-2</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing philentropy</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Hajk-Georg Drost <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-1567-306X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Hajk-Georg Drost.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer>
</div>






  </body>
</html>
