<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Similarity and Distance Quantification Between Probability Functions • philentropy</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Similarity and Distance Quantification Between Probability Functions">
<meta property="og:description" content="Computes 46 optimized distance and similarity measures for comparing probability functions (Drost (2018) &lt;doi:10.21105/joss.00765&gt;). These comparisons between probability functions have their foundations in a broad range of scientific disciplines from mathematics to ecology. The aim of this package is to provide a core framework for clustering, classification, statistical inference, goodness-of-fit, non-parametric statistics, information theory, and machine learning tasks that are based on comparing univariate or multivariate probability functions.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">philentropy</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.8.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/Distances.html">Distances</a>
    </li>
    <li>
      <a href="articles/Information_Theory.html">Information Theory</a>
    </li>
    <li>
      <a href="articles/Introduction.html">Introduction</a>
    </li>
    <li>
      <a href="articles/Many_Distances.html">Comparing many probability density functions</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drostlab/philentropy/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="philentropy">philentropy<a class="anchor" aria-label="anchor" href="#philentropy"></a>
</h1></div>
<p><a href="https://travis-ci.org/HajkD/philentropy" class="external-link"><img src="https://travis-ci.org/HajkD/philentropy.svg?branch=master" alt="Travis-CI Build Status"></a> <a href="https://joss.theoj.org/papers/10.21105/joss.00765" class="external-link"><img src="http://joss.theoj.org/papers/cad5ffc246ce197b06ccad1af7d2932a/status.svg" alt="status"></a> <a href="https://github.com/r-hub/cranlogs.app" class="external-link"><img src="http://cranlogs.r-pkg.org/badges/philentropy" alt="rstudio mirror downloads"></a> <a href="https://github.com/r-hub/cranlogs.app" class="external-link"><img src="http://cranlogs.r-pkg.org/badges/grand-total/philentropy" alt="rstudio mirror downloads"></a></p>
<div class="section level3">
<h3 id="similarity-and-distance-quantification-between-probability-functions">Similarity and Distance Quantification between Probability Functions<a class="anchor" aria-label="anchor" href="#similarity-and-distance-quantification-between-probability-functions"></a>
</h3>
<blockquote>
<p>Describe and understand the world through data.</p>
</blockquote>
<p>Data collection and data comparison are the foundations of scientific research. <em>Mathematics</em> provides the abstract framework to describe patterns we observe in nature and <em>Statistics</em> provides the framework to quantify the uncertainty of these patterns. In statistics, natural patterns are described in form of probability distributions which either follow a fixed pattern (parametric distributions) or more dynamic patterns (non-parametric distributions).</p>
<p>The <code>philentropy</code> package implements fundamental distance and similarity measures to quantify distances between probability density functions as well as traditional information theory measures. In this regard, it aims to provide a framework for comparing natural patterns in a statistical notation.</p>
<p>This project is born out of my passion for statistics and I hope that it will be useful to the people who share it with me.</p>
</div>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install philentropy version 0.7.0 from CRAN</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"philentropy"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h3>
<p><strong>I am developing <code>philentropy</code> in my spare time and would be very grateful if you would consider citing the following paper in case <code>philentropy</code> was useful for your own research. I plan on maintaining and extending the <code>philentropy</code> functionality and usability in the next years and require citations to back up these efforts. Many thanks in advance :)</strong></p>
<blockquote>
<p>HG Drost, (2018). <strong>Philentropy: Information Theory and Distance Quantification with R</strong>. <em>Journal of Open Source Software</em>, 3(26), 765. <a href="https://doi.org/10.21105/joss.00765" class="external-link uri">https://doi.org/10.21105/joss.00765</a></p>
</blockquote>
</div>
<div class="section level2">
<h2 id="tutorials">Tutorials<a class="anchor" aria-label="anchor" href="#tutorials"></a>
</h2>
<ul>
<li><a href="https://drostlab.github.io/philentropy/articles/Introduction.html" class="external-link">Introduction to the philentropy package</a></li>
<li><a href="https://drostlab.github.io/philentropy/articles/Distances.html" class="external-link">Distance and Similarity Measures implemented in philentropy</a></li>
<li><a href="https://drostlab.github.io/philentropy/articles/Information_Theory.html" class="external-link">Information Theory Metrics implemented in philentropy</a></li>
<li><a href="https://drostlab.github.io/philentropy/articles/Many_Distances.html" class="external-link">Comparing many probability density functions</a></li>
</ul>
</div>
<div class="section level2">
<h2 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drostlab/philentropy" class="external-link">philentropy</a></span><span class="op">)</span></span>
<span><span class="co"># retrieve available distance metrics</span></span>
<span><span class="fu">philentropy</span><span class="fu">::</span><span class="fu"><a href="reference/getDistMethods.html">getDistMethods</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code> [1] "euclidean"         "manhattan"         "minkowski"        
 [4] "chebyshev"         "sorensen"          "gower"            
 [7] "soergel"           "kulczynski_d"      "canberra"         
[10] "lorentzian"        "intersection"      "non-intersection" 
[13] "wavehedges"        "czekanowski"       "motyka"           
[16] "kulczynski_s"      "tanimoto"          "ruzicka"          
[19] "inner_product"     "harmonic_mean"     "cosine"           
[22] "hassebrook"        "jaccard"           "dice"             
[25] "fidelity"          "bhattacharyya"     "hellinger"        
[28] "matusita"          "squared_chord"     "squared_euclidean"
[31] "pearson"           "neyman"            "squared_chi"      
[34] "prob_symm"         "divergence"        "clark"            
[37] "additive_symm"     "kullback-leibler"  "jeffreys"         
[40] "k_divergence"      "topsoe"            "jensen-shannon"   
[43] "jensen_difference" "taneja"            "kumar-johnson"    
[46] "avg"</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># define a probability density function P</span></span>
<span><span class="va">P</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="co"># define a probability density function Q</span></span>
<span><span class="va">Q</span> <span class="op">&lt;-</span> <span class="fl">20</span><span class="op">:</span><span class="fl">29</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fl">20</span><span class="op">:</span><span class="fl">29</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># combine P and Q as matrix object</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">P</span>,<span class="va">Q</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># compute the jensen-shannon distance between</span></span>
<span><span class="co"># probability density functions P and Q</span></span>
<span><span class="fu">philentropy</span><span class="fu">::</span><span class="fu"><a href="reference/distance.html">distance</a></span><span class="op">(</span><span class="va">x</span>, method <span class="op">=</span> <span class="st">"jensen-shannon"</span><span class="op">)</span></span></code></pre></div>
<pre><code>jensen-shannon using unit 'log'.
jensen-shannon 
    0.02628933</code></pre>
<p>Alternatively, users can also retrieve values from all available distance/similarity metrics using <code><a href="reference/dist.diversity.html">philentropy::dist.diversity()</a></code>:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">philentropy</span><span class="fu">::</span><span class="fu"><a href="reference/dist.diversity.html">dist.diversity</a></span><span class="op">(</span><span class="va">x</span>, p <span class="op">=</span> <span class="fl">2</span>, unit <span class="op">=</span> <span class="st">"log2"</span><span class="op">)</span></span></code></pre></div>
<pre><code>        euclidean         manhattan 
       0.12807130        0.35250464 
        minkowski         chebyshev 
       0.12807130        0.06345083 
         sorensen             gower 
       0.17625232        0.03525046 
          soergel      kulczynski_d 
       0.29968454        0.42792793 
         canberra        lorentzian 
       2.09927095        0.49712136 
     intersection  non-intersection 
       0.82374768        0.17625232 
       wavehedges       czekanowski 
       3.16657887        0.17625232 
           motyka      kulczynski_s 
       0.58812616        2.33684211 
         tanimoto           ruzicka 
       0.29968454        0.70031546 
    inner_product     harmonic_mean 
       0.10612245        0.94948528 
           cosine        hassebrook 
       0.93427641        0.86613103 
          jaccard              dice 
       0.13386897        0.07173611 
         fidelity     bhattacharyya 
       0.97312397        0.03930448 
        hellinger          matusita 
       0.32787819        0.23184489 
    squared_chord squared_euclidean 
       0.05375205        0.01640226 
          pearson            neyman 
       0.16814418        0.36742465 
      squared_chi         prob_symm 
       0.10102943        0.20205886 
       divergence             clark 
       1.49843905        0.86557468 
    additive_symm  kullback-leibler 
       0.53556883        0.13926288 
         jeffreys      k_divergence 
       0.31761069        0.04216273 
           topsoe    jensen-shannon 
       0.07585498        0.03792749 
jensen_difference            taneja 
       0.03792749        0.04147518 
    kumar-johnson               avg 
       0.62779644        0.20797774</code></pre>
<div class="section level3">
<h3 id="install-developer-version">Install Developer Version<a class="anchor" aria-label="anchor" href="#install-developer-version"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="co"># install the current version of philentropy on your system</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"HajkD/philentropy"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="news">NEWS<a class="anchor" aria-label="anchor" href="#news"></a>
</h3>
<p>The current status of the package as well as a detailed history of the functionality of each version of <code>philentropy</code> can be found in the <a href="https://drostlab.github.io/philentropy/news/index.html" class="external-link">NEWS</a> section.</p>
</div>
</div>
<div class="section level2">
<h2 id="important-functions">Important Functions<a class="anchor" aria-label="anchor" href="#important-functions"></a>
</h2>
<div class="section level3">
<h3 id="distance-measures">Distance Measures<a class="anchor" aria-label="anchor" href="#distance-measures"></a>
</h3>
<ul>
<li>
<code><a href="reference/distance.html">distance()</a></code> : Implements 46 fundamental probability distance (or similarity) measures</li>
<li>
<code><a href="reference/getDistMethods.html">getDistMethods()</a></code> : Get available method names for ‘distance’</li>
<li>
<code><a href="reference/dist.diversity.html">dist.diversity()</a></code> : Distance Diversity between Probability Density Functions</li>
<li>
<code><a href="reference/estimate.probability.html">estimate.probability()</a></code> : Estimate Probability Vectors From Count Vectors</li>
</ul>
</div>
<div class="section level3">
<h3 id="information-theory">Information Theory<a class="anchor" aria-label="anchor" href="#information-theory"></a>
</h3>
<ul>
<li>
<code><a href="reference/H.html">H()</a></code> : Shannon’s Entropy H(X)</li>
<li>
<code><a href="reference/JE.html">JE()</a></code> : Joint-Entropy H(X,Y)</li>
<li>
<code><a href="reference/CE.html">CE()</a></code> : Conditional-Entropy H(X | Y)</li>
<li>
<code><a href="reference/MI.html">MI()</a></code> : Shannon’s Mutual Information I(X,Y)</li>
<li>
<code><a href="reference/KL.html">KL()</a></code> : Kullback–Leibler Divergence</li>
<li>
<code><a href="reference/JSD.html">JSD()</a></code> : Jensen-Shannon Divergence</li>
<li>
<code><a href="reference/gJSD.html">gJSD()</a></code> : Generalized Jensen-Shannon Divergence</li>
</ul>
</div>
<div class="section level3">
<h3 id="studies-that-successfully-applied-the-philentropy-package">Studies that successfully applied the <code>philentropy</code> package<a class="anchor" aria-label="anchor" href="#studies-that-successfully-applied-the-philentropy-package"></a>
</h3>
<blockquote>
<ul>
<li><p><strong>Annelid functional genomics reveal the origins of bilaterian life cycles</strong> FM Martín-Zamora, Y Liang, K Guynes et al.- <strong>Nature</strong>, 2023</p></li>
<li><p><strong>An atlas of gene regulatory elements in adult mouse cerebrum</strong> YE Li, S Preissl, X Hou, Z Zhang, K Zhang et al.- <strong>Nature</strong>, 2021</p></li>
<li><p><strong>Convergent somatic mutations in metabolism genes in chronic liver disease</strong> S Ng, F Rouhani, S Brunner, N Brzozowska et al. <strong>Nature</strong>, 2021</p></li>
<li><p><strong>Antigen dominance hierarchies shape TCF1+ progenitor CD8 T cell phenotypes in tumors</strong> ML Burger, AM Cruz, GE Crossland et al. - <strong>Cell</strong>, 2021</p></li>
<li><p><strong>High-content single-cell combinatorial indexing</strong> R Mulqueen et al. - <strong>Nature Biotechnology</strong>, 2021</p></li>
<li><p><strong>A comparative atlas of single-cell chromatin accessibility in the human brain</strong> YE Li, S Preissl, M Miller, ND Johnson, Z Wang et al. - <strong>Science</strong>, 2023</p></li>
<li><p><strong>Extinction at the end-Cretaceous and the origin of modern Neotropical rainforests</strong> MR Carvalho, C Jaramillo et al. - <strong>Science</strong>, 2021</p></li>
<li><p><strong>sciCSR infers B cell state transition and predicts class-switch recombination dynamics using single-cell transcriptomic data</strong> JCF Ng, G Montamat Garcia, AT Stewart et al. - <strong>Nature Methods</strong>, 2023</p></li>
<li><p><strong>HERMES: a molecular-formula-oriented method to target the metabolome</strong> R Giné, J Capellades, JM Badia et al. - <strong>Nature Methods</strong>, 2021</p></li>
<li><p><strong>The genetic architecture of temperature adaptation is shaped by population ancestry and not by selection regime</strong> KA Otte, V Nolte, F Mallard et al. - <strong>Genome Biology</strong>, 2021</p></li>
<li><p><strong>The Tug1 lncRNA locus is essential for male fertility</strong> JP Lewandowski et al. - <strong>Genome Biology</strong>, 2020</p></li>
<li><p><strong>Transcriptional vulnerabilities of striatal neurons in human and rodent models of Huntington’s disease</strong> A Matsushima, SS Pineda, JR Crittenden et al. - <strong>Nature Communications</strong>, 2023</p></li>
<li><p><strong>Resolving the structure of phage–bacteria interactions in the context of natural diversity</strong> KM Kauffman, WK Chang, JM Brown et al. - <strong>Nature Communications</strong>, 2022</p></li>
<li><p><strong>Gut microbiome-mediated metabolism effects on immunity in rural and urban African populations</strong> M Stražar, GS Temba, H Vlamakis et al. - <strong>Nature Communications</strong>, 2021</p></li>
<li><p><strong>Aging, inflammation and DNA damage in the somatic testicular niche with idiopathic germ cell aplasia</strong> M Alfano, AS Tascini, F Pederzoli et al. - <strong>Nature communications</strong>, 2021</p></li>
<li><p><strong>Single cell census of human kidney organoids shows reproducibility and diminished off-target cells after transplantation</strong> A Subramanian et al. - <strong>Nature Communications</strong>, 2019</p></li>
<li><p><strong>A single-cell multi-omic atlas spanning the adult rhesus macaque brain</strong> KL Chiou, X Huang, MO Bohlen et al. - <strong>Science Advances</strong>, 2023</p></li>
<li><p><strong>Different languages, similar encoding efficiency: Comparable information rates across the human communicative niche</strong> C Coupé, YM Oh, D Dediu, F Pellegrino - <strong>Science Advances</strong>, 2019</p></li>
<li><p><strong>Single-cell deletion analyses show control of pro–T cell developmental speed and pathways by Tcf7, Spi1, Gata3, Bcl11a, Erg, and Bcl11b</strong> W Zhou, F Gao, M Romero-Wolf, S Jo, EV Rothenberg - <strong>Science Immunology</strong>, 2022</p></li>
<li><p><strong>Large-scale iterated singing experiments reveal oral transmission mechanisms underlying music evolution</strong> M Anglada-Tort, PMC Harrison, H Lee, N Jacoby - <strong>Current Biology</strong>, 2023</p></li>
<li><p><strong>TAS-Seq is a robust and sensitive amplification method for bead-based scRNA-seq</strong> S Shichino, S Ueha, S Hashimoto, T Ogawa et al. - <strong>Communications biology</strong>, 2022</p></li>
<li><p>Mapping hormone-regulated cell-cell interaction networks in the human breast at single-cell resolution LM Murrow, RJ Weber, JA Caruso et al. - <strong>Cell Systems</strong>, 2022</p></li>
<li><p><strong>Large-scale chromatin reorganization reactivates placenta-specific genes that drive cellular aging</strong> Z Liu, Q Ji, J Ren, P Yan, Z Wu, S Wang, L Sun, Z Wang et al. - <strong>Developmental Cell</strong>, 2022</p></li>
<li><p><strong>Direct epitranscriptomic regulation of mammalian translation initiation through N4-acetylcytidine</strong> D Arango, D Sturgill, R Yang, T Kanai, P Bauer et al. - <strong>Molecular Cell</strong>, 2022</p></li>
<li><p><strong>Loss of adaptive capacity in asthmatic patients revealed by biomarker fluctuation dynamics after rhinovirus challenge</strong> A Sinha et al. - <strong>eLife</strong>, 2019</p></li>
<li><p><strong>Sex and hatching order modulate the association between MHC‐II diversity and fitness in early‐life stages of a wild seabird</strong> M Pineaux et al - <strong>Molecular Ecology</strong>, 2020</p></li>
<li><p><strong>BELMM: Bayesian model selection and random walk smoothing in time-series clustering</strong> O Sarala, T Pyhäjärvi, MJ Sillanpää - <strong>Bioinformatics</strong>, 2023</p></li>
<li><p><strong>Cellsig plug-in enhances CIBERSORTx signature selection for multi-dataset transcriptomes with sparse multilevel modelling</strong> MA Al Kamran Khan, J Wu, S Yuhan et al. - <strong>Bioinformatics</strong>, 2023</p></li>
<li><p><strong>How the Choice of Distance Measure Influences the Detection of Prior-Data Conflict</strong> K Lek, R Van De Schoot - <strong>Entropy</strong>, 2019</p></li>
<li><p><strong>Differential variation analysis enables detection of tumor heterogeneity using single-cell RNA-sequencing data</strong> EF Davis-Marcisak, TD Sherman et al. - <strong>Cancer research</strong>, 2019</p></li>
<li><p><strong>Multi-Omics Investigation of Innate Navitoclax Resistance in Triple-Negative Breast Cancer Cells</strong> M Marczyk et al. - <strong>Cancers</strong>, 2020</p></li>
<li><p><strong>Impact of Gut Microbiome on Hypertensive Patients with Low-Salt Intake: Shika Study Results</strong> S Nagase et al. - <strong>Frontiers in Medicine</strong>, 2020</p></li>
<li><p>Children’s social networks in developmental psychology: A network approach to capture and describe early social environments N Burke, N Brezack, A Woodward - <strong>Frontiers in psychology</strong>, 2022</p></li>
<li><p><strong>Combined TCR Repertoire Profiles and Blood Cell Phenotypes Predict Melanoma Patient Response to Personalized Neoantigen Therapy plus Anti-PD-1</strong> A Poran et al. - <strong>Cell Reports Medicine</strong>, 2020</p></li>
<li><p><strong>Identification of a glioma functional network from gene fitness data using machine learning</strong> C Xiang, X Liu, D Zhou, Y Zhou, X Wang, F Chen - <strong>Journal of Cellular and Molecular Medicine</strong>, 2022</p></li>
<li><p><strong>Whole genome assemblies of Zophobas morio and Tenebrio molitor</strong> S Kaur, SA Stinson, GC diCenzo - <strong>G3: Genes, Genomes, Genetics</strong>, 2023</p></li>
<li><p><strong>Prediction of New Risk Genes and Potential Drugs for Rheumatoid Arthritis from Multiomics Data</strong> AM Birga, L Ren, H Luo, Y Zhang, J Huang - <strong>Computational and Mathematical Methods in Medicine</strong>, 2022</p></li>
<li><p><strong>Phenotyping of acute and persistent COVID-19 features in the outpatient setting: exploratory analysis of an international cross-sectional online survey</strong> S Sahanic, P Tymoszuk, D Ausserhofer et al. - <strong>medRxiv</strong>, 2021</p></li>
<li><p><strong>A two-part evaluation approach for measuring the usability and user experience of an Augmented Reality-based assistance system to support the temporal coordination of spatially dispersed teams</strong> L Thomaschewski, B Weyers, A Kluge - <strong>Cognitive Systems Research</strong>, 2021</p></li>
<li><p><strong>SEDE-GPS: socio-economic data enrichment based on GPS information</strong> T Sperlea, S Füser, J Boenigk, D Heider - <strong>BMC bioinformatics</strong>, 2018</p></li>
<li><p><strong>Longitudinal analysis on the ecological dynamics of the cervicovaginal microbiome in hrHPV infection</strong> MA Molina, WJG Melchers, et al. - <strong>Computational and structural biotechnology</strong>, 2023</p></li>
<li><p><strong>Spatial and molecular anatomy of germ layers in the gastrulating primate embryo</strong> G Cui, S Feng, Y Yan, L Wang, X He, X Li, et al. - <strong>bioRxiv</strong>, 2022</p></li>
<li><p><strong>Evacuees and Migrants Exhibit Different Migration Systems after the Great East Japan Earthquake and Tsunami</strong> M Hauer, S Holloway, T Oda – 2019</p></li>
<li><p><strong>Robust comparison of similarity measures in analogy based software effort estimation</strong> P Phannachitta - <strong>11th International Conference on Software</strong>, 2017</p></li>
<li><p><strong>RUNIMC - An R-based package for imaging mass cytometry data analysis and pipeline validation</strong> L Dolcetti, PR Barber, G Weitsman, S Thavarajet al. - <strong>bioRxiv</strong>, 2021</p></li>
<li><p><strong>Expression variation analysis for tumor heterogeneity in single-cell RNA-sequencing data</strong> EF Davis-Marcisak, P Orugunta et al. - <strong>BioRxiv</strong>, 2018</p></li>
<li><p><strong>Concept acquisition and improved in-database similarity analysis for medical data</strong> I Wiese, N Sarna, L Wiese, A Tashkandi, U Sax - <strong>Distributed and Parallel Databases</strong>, 2019</p></li>
<li><p><strong>Dynamics of Vaginal and Rectal Microbiota over Several Menstrual Cycles in Female Cynomolgus Macaques</strong> MT Nugeyre, N Tchitchek, C Adapen et al. - <strong>Frontiers in Cellular and Infection Microbiology</strong>, 2019</p></li>
<li><p><strong>Inferring the quasipotential landscape of microbial ecosystems with topological data analysis</strong> WK Chang, L Kelly - <strong>BioRxiv</strong>, 2019</p></li>
<li><p><strong>Shifts in the nasal microbiota of swine in response to different dosing regimens of oxytetracycline administration</strong> KT Mou, HK Allen, DP Alt, J Trachsel et al. - <strong>Veterinary microbiology</strong>, 2019</p></li>
<li><p><strong>The Patchy Distribution of Restriction–Modification System Genes and the Conservation of Orphan Methyltransferases in Halobacteria</strong> MS Fullmer, M Ouellette, AS Louyakis et al. - <strong>Genes</strong>, 2019</p></li>
<li><p><strong>Genetic differentiation and intrinsic genomic features explain variation in recombination hotspots among cocoa tree populations</strong> EJ Schwarzkopf, JC Motamayor, OE Cornejo - <strong>BioRxiv</strong>, 2019</p></li>
<li><p><strong>Metastable regimes and tipping points of biochemical networks with potential applications in precision medicine</strong> SS Samal, J Krishnan, AH Esfahani et al. - <strong>Reasoning for Systems Biology and Medicine</strong>, 2019</p></li>
<li><p><strong>Genome‐wide characterization and developmental expression profiling of long non‐coding RNAs in Sogatella furcifera</strong> ZX Chang, OE Ajayi, DY Guo, QF Wu - <strong>Insect science</strong>, 2019</p></li>
<li><p><strong>Development of a simulation system for modeling the stock market to study its characteristics</strong> P Mariya – 2018</p></li>
<li><p><strong>The Tug1 Locus is Essential for Male Fertility</strong> JP Lewandowski, G Dumbović, AR Watson, T Hwang et al. - <strong>BioRxiv</strong>, 2019</p></li>
<li><p><strong>Microbiotyping the sinonasal microbiome</strong> A Bassiouni, S Paramasivan, A Shiffer et al. - <strong>BioRxiv</strong>, 2019</p></li>
<li><p><strong>Critical search: A procedure for guided reading in large-scale textual corpora</strong> J Guldi - <strong>Journal of Cultural Analytics</strong>, 2018</p></li>
<li><p><strong>A Bibliography of Publications about the R, S, and S-Plus Statistics Programming Languages</strong> NHF Beebe – 2019</p></li>
<li><p><strong>Improved state change estimation in dynamic functional connectivity using hidden semi-Markov models</strong> H Shappell, BS Caffo, JJ Pekar, MA Lindquist - <strong>NeuroImage</strong>, 2019</p></li>
<li><p><strong>A Smart Recommender Based on Hybrid Learning Methods for Personal Well-Being Services</strong> RM Nouh, HH Lee, WJ Lee, JD Lee - <strong>Sensors</strong>, 2019</p></li>
<li><p><strong>Cognitive Structural Accuracy</strong> V Frenz – 2019</p></li>
<li><p><strong>Kidney organoid reproducibility across multiple human iPSC lines and diminished off target cells after transplantation revealed by single cell transcriptomics</strong> A Subramanian, EH Sidhom, M Emani et al. - <strong>BioRxiv</strong>, 2019</p></li>
<li><p><strong>Multi-classifier majority voting analyses in provenance studies on iron artefacts</strong> G Żabiński et al. - <strong>Journal of Archaeological Science</strong>, 2020</p></li>
<li><p><strong>Identifying inhibitors of epithelial–mesenchymal plasticity using a network topology-based approach</strong> K Hari et al. - <strong>NPJ systems biology and applications</strong>, 2020</p></li>
<li><p><strong>Genetic differentiation and intrinsic genomic features explain variation in recombination hotspots among cocoa tree populations</strong> EJ Schwarzkopf et al. - <strong>BMC Genomics</strong>, 2020</p></li>
<li><p><strong>Enhancing Card Sorting Dendrograms through the Holistic Analysis of Distance Methods and Linkage Criteria.</strong> JA Macías - <strong>Journal of Usability Studies</strong>, 2021</p></li>
<li><p><strong>Pattern-based identification and mapping of landscape types using multi-thematic data</strong> J Nowosad, TF Stepinski - <strong>International Journal of Geographical Information</strong>, 2021</p></li>
<li><p><strong>Motif Analysis in k-mer Networks: An Approach towards Understanding SARS-CoV-2 Geographical Shifts</strong> S Biswas, S Saha, S Bandyopadhyay, M Bhattacharyya - <strong>bioRxiv</strong>, 2020</p></li>
<li><p><strong>Motif: an open-source R tool for pattern-based spatial analysis</strong> J Nowosad - <strong>Landscape Ecology</strong>, 2021</p></li>
<li><p><strong>New effective spectral matching measures for hyperspectral data analysis</strong> C Kumar, S Chatterjee, T Oommen, A Guha - <strong>International Journal of Remote Sensing</strong>, 2021</p></li>
<li><p><strong>Innovative activity of Polish enterprises–a strategic aspect. The similarity of NACE divisions</strong> E Bielińska-Dusza, M Hamerska - <strong>Journal of Entrepreneurship, Management and innovation</strong>, 2021</p></li>
<li><p><strong>Multi-classifier majority voting analyses in provenance studies on iron artefacts</strong> G Żabiński, J Gramacki et al.- <strong>Journal of Archaeological Science</strong>, 2020</p></li>
<li><p><strong>Unraveling the record of a tropical continental Cretaceous-Paleogene boundary in northern Colombia, South America</strong> F de la Parra, C Jaramillo, P Kaskes et al. - <strong>Journal of South American Earth Sciences</strong>, 2022</p></li>
<li><p><strong>A roadmap to reconstructing muscle architecture from CT data</strong> J Katzke, P Puchenkov, H Stark, EP Economo - <strong>Integrative Organismal Biology</strong>, 2022</p></li>
<li><p><strong>Pandemonium: a clustering tool to partition parameter space—application to the B anomalies</strong> U Laa, G Valencia - <strong>The European Physical Journal Plus</strong>, 2022</p></li>
<li><p><strong>Identification of a glioma functional network from gene fitness data using machine learning</strong> C Xiang, X Liu, D Zhou, Y Zhou, X Wang, F Chen - <strong>Journal of Cellular and Molecular Medicine</strong>, 2022</p></li>
<li><p><strong>Cross compatibility in intraspecific and interspecific hybridization in yam (Dioscorea spp.)</strong> JM Mondo, PA Agre, A Edemodu et al. - <strong>Scientific reports</strong>, 2022</p></li>
<li><p><strong>A Modular and Expandable Ecosystem for Metabolomics Data Annotation in R</strong> J Rainer, A Vicini, L Salzer, J Stanstrup et al. - <strong>Metabolites</strong>, 2022</p></li>
<li><p><strong>Single-Cell Transcriptome Integration Analysis Reveals the Correlation Between Mesenchymal Stromal Cells and Fibroblasts</strong> C Fan, M Liao, L Xie, L Huang, S Lv, S Cai et al. - <strong>Frontiers in genetics</strong>, 2022</p></li>
<li><p><strong>Phenotypic regionalization of the vertebral column in the thorny skate Amblyraja radiata: Stability and variation</strong> F Berio, Y Bayle, C Riley, O Larouche, R Cloutier - <strong>Journal of Anatomy</strong>, 2022</p></li>
<li><p><strong>Community assembly during vegetation succession after metal mining is driven by multiple processes with temporal variation</strong> T Li, H Yang, X Yang, Z Guo, D Fu, C Liu, S Li et al. - <strong>Ecology and evolution</strong>, 2022</p></li>
<li><p><strong>Integrative Organismal Biology</strong> J Katzke, P Puchenkov, H Stark, <strong>EP Economo</strong> - 2022</p></li>
<li><p><strong>Optimizing use of US Ex-PVP inbred lines for enhancing agronomic performance of tropical Striga resistant maize inbred lines</strong> ARS Maazou, M Gedil, VO Adetimirin et al. - <strong>BMC Plant Biology</strong>, 2022</p></li>
</ul>
</blockquote>
</div>
</div>
<div class="section level2">
<h2 id="discussions-and-bug-reports">Discussions and Bug Reports<a class="anchor" aria-label="anchor" href="#discussions-and-bug-reports"></a>
</h2>
<p>I would be very happy to learn more about potential improvements of the concepts and functions provided in this package.</p>
<p>Furthermore, in case you find some bugs or need additional (more flexible) functionality of parts of this package, please let me know:</p>
<p><a href="https://github.com/drostlab/philentropy/issues" class="external-link uri">https://github.com/drostlab/philentropy/issues</a></p>
<p>or find me on <a href="https://twitter.com/hajkdrost" class="external-link">twitter: HajkDrost</a></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=philentropy" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/drostlab/philentropy/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/drostlab/philentropy/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-2" class="external-link">GPL-2</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing philentropy</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Hajk-Georg Drost <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/http://orcid.org/0000-0002-1567-306X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Hajk-Georg Drost.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
